File: app/src/main/java/com/chebuso/chargetimer/notifications/CarChargedAlarmReceiver.java
package com.chebuso.chargetimer.notifications;

import android.app.Notification;
import android.app.NotificationManager;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class CarChargedAlarmReceiver extends BroadcastReceiver
{
    public static final String NOTIFICATION_ID = "car-charged-notification-id";
    public static final String NOTIFICATION = "notification";

    @Override
    public void onReceive(Context context, Intent intent) {
        NotificationManager notificationManager = (NotificationManager)context.getSystemService(Context.NOTIFICATION_SERVICE);
        if (notificationManager != null) {
            Notification notification = intent.getParcelableExtra(NOTIFICATION);
            int id = intent.getIntExtra(NOTIFICATION_ID, 0);
            notificationManager.notify(id, notification);
        }
    }
}

File: EvChargeTimer/app/src/main/java/com/chebuso/chargetimer/notifications/CarChargedAlarmReceiver.kt
package com.chebuso.chargetimer.notifications

import android.app.Notification
import android.app.NotificationManager
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log


class CarChargedAlarmReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        Log.d(TAG, "onReceive")
        val notificationManager =
            context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
        val notification = intent.getParcelableExtra<Notification>(NOTIFICATION)
        val id = intent.getIntExtra(NOTIFICATION_ID, 0)
        notificationManager.notify(id, notification)
        Log.i(TAG, "Sent notification to the user")
    }

    companion object {
        const val NOTIFICATION_ID = "car-charged-notification-id"
        const val NOTIFICATION = "notification"
        private val TAG = CarChargedAlarmReceiver::class.java.simpleName
    }
}

